lib/charms/openfga_k8s/v1/openfga.py

```diff
@@ -0,0 +1,423 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""# Interface Library for OpenFGA.
+
+This library wraps relation endpoints using the `openfga` interface
+and provides a Python API for requesting OpenFGA authorization model
+stores to be created.
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.openfga_k8s.v1.openfga
+```
+
+In the `metadata.yaml` of the charm, add the following:
+
+```yaml
+requires:
+  openfga:
+    interface: openfga
+```
+
+Then, to initialise the library:
+```python
+from charms.openfga_k8s.v1.openfga import (
+    OpenFGARequires,
+    OpenFGAStoreCreateEvent,
+)
+
+class SomeCharm(CharmBase):
+  def __init__(self, *args):
+    # ...
+    self.openfga = OpenFGARequires(self, "test-openfga-store")
+    self.framework.observe(
+        self.openfga.on.openfga_store_created,
+        self._on_openfga_store_created,
+    )
+
+    def _on_openfga_store_created(self, event: OpenFGAStoreCreateEvent):
+        if not self.unit.is_leader():
+            return
+
+        if not event.store_id:
+            return
+
+        info = self.openfga.get_store_info()
```

Comment:
why is only the leader allowed to get the store info? (looking at line 44)

Reply:
since this is a read operation, i think all units should be allowed to get_store_info

Reply:
Fair